src/test/cpp/test/CmdReader.cpp

```diff
@@ -0,0 +1,137 @@
+#include "pch.h"
+#include "winrt/Windows.Foundation.h"
+#include "cmd_reader.h"
+#include <fstream>
+
+using namespace xlang::cmd;
+
+TEST_CASE("CmdReader")
+{
+    const std::vector<option> options
+    {
+        { "input", 1 },
+        { "reference", 0 },
+        { "output", 0, 1 },
+        { "component", 0, 1 },
+        { "filter", 0 },
+        { "name", 0, 1 },
+        { "verbose", 0, 0 },
+    };
+
+    // input and output
+    {
+        const char* argv[] = { "progname", "-in", "example_file.in", "-out", "example_file.out" };
+        const size_t argc = 5;
+        reader args{ argc, argv, options };
+
+        REQUIRE(args.exists("input"));
+        REQUIRE(args.value("input") == "example_file.in");
+        REQUIRE_FALSE(args.exists("reference"));
+        REQUIRE(args.exists("output"));
+        REQUIRE(args.value("output") == "example_file.out");
+        REQUIRE_FALSE(args.exists("filter"));
+        REQUIRE_FALSE(args.exists("name"));
+        REQUIRE_FALSE(args.exists("verbose"));
+    }
+
+    // response file #1: filename no quotes
+    {
+        const char* argv[] = { "progname", "@respfile.txt" };
+        const size_t argc = 2;
+        std::ofstream resp_file;
```

Comment:
consider making an raii type here to consolidate common logic and delete file in presence of an exception

Reply:
Might be simpler and more maintainable just to have an actual file in the test project containing the test arguments and not bother with generating and then deleting a file.

Reply:
I actually like the generation code - everything in one place